sips/sip-x/sip-x-improved-stacking.md

```diff
@@ -0,0 +1,205 @@
+- stack-extend
+
+## Locking Post Conditions
+
+The following type of post conditions shall be added to the current definition in SIP-005 (`TransactionPostCondition`)
+
+- `LockingLimit(PostConditionPrincipal, FungibleConditionCode, u64)`
+
+A transaction (using Deny mode) with this post condition will abort if the locked Stacks tokens of the principal does not satisfy the provided conditions. The logic for the conditions follows that of STX transfer.
+
+## Automatic Extend and Locking period
+
+The following function shall be removed:
+
+- `delegate-stack-extend`
+
+In addition, the stacking settings for each stacker (user with delegation) shall be applied for the next stacking cycle if the user did not signal the end of stacking by calling `revoke-delegate-stx` 200 blocks before the start of the next cycle.
+
+This results in the following structure of the stacking cycle:
+
+- bitcoin block 1-1900: stacking as usual, user can signal change of stacking settings.
+- bitcoin block 1901-2000: stacking as usual, signers can still aggregate stacking changes.
+- bitcoin block 2001-2100 (prepare phase): no rewards for stackers, changes to stacking are applied to the cycle after next.
+
+```mermaid
+timeline
+    title Stacking Cycle Structure
+    section  Earning
+    1 : Stacking as usual : User can signal change of stacking settings
+    ... : Usual reward distribution
+    1901 : Stacking as usual : Auto extension :  Signers can still aggregate stacking changes
+    section No earning
+    2001 : Prepare phase, no rewards: delegate txs applied only to cycle after next
+    2100 : End of cycle
+```
+
+That means the locking period is 1 cycle, with automatic extension for another cycle until the user decides to end stacking.
+
+## Semantic Change of Delegation
+
+When signers verify and accept proposed blocks by miners, their voting power corresponds to the amount of stacked Stacks tokens (see SIP-021). The new stacking process changes the delegation flow as follows:
+
+- `delegate-stx` shall be renamed to `delegate`. The function takes the arguments: `amount`, `delegate-to`, `signature`, optional bitcoin `block-height` defining the end of stacking and auto extending, optional `max-amount`. The `amount` defines how many Stacks tokens are locked immediately. The `max-amount` the maximum of Stacks tokens that can be locked in the future through auto extending. If omitted the whole balance is locked. The argument `delegate-to` describes the signer. The argument `signature` is a signature of the signer indicating that the signer accepted the delegation of voting power.
+
+- `revoke-delegate-stx` shall be renamed to `revoke-delegate`. After calling this functions, auto extension is stopped and Stacks tokens are unlocked for the user at the end of the current cycle. If the signer the user delegated to did not aggregated enough Stacks tokens to receive at least one reward slot then the user's Stacks token are unlocked immediately through this call.
```

Comment:
...did not aggregate enough...

(instead of _aggregated_)

Reply:
:+1: